File: Beginner/UMGForCs/Source/UMGForCs/Game.Generated/UMGForCsGameMode.cpp
#include "UMGForCsGameMode.h"
void AUMGForCsGameMode::BeginPlay()
{
Super::BeginPlay();
ChangeMenuWidget(StartingWidgetClass);
}
void AUMGForCsGameMode::ChangeMenuWidget(TSubclassOf<UUserWidget> NewWidgetClass)
{
if (CurrentWidget!=nullptr)
{
CurrentWidget->RemoveFromViewport();
CurrentWidget=nullptr;
}
if (NewWidgetClass!=nullptr)
{
CurrentWidget=CreateWidget<UUserWidget>(GetWorld(),NewWidgetClass,TEXT(""));
if (CurrentWidget!=nullptr)
{
CurrentWidget->AddToViewport(0);
}
}
}

File: Beginner/UMGForCs/Source/UMGForCs/Game/UMGForCsGameMode.cs
namespace Game;

using GameFramework;
using System.Diagnostics.CodeAnalysis;

public class AUMGForCsGameMode : AGameModeBase
{
    TSubclassOf<UUserWidget> StartingWidgetClass;

    [AllowNull]
    UUserWidget CurrentWidget;

    protected override void BeginPlay()
    {
        base.BeginPlay();
        ChangeMenuWidget(StartingWidgetClass);
    }

    void ChangeMenuWidget(TSubclassOf<UUserWidget> NewWidgetClass)
    {
        if (CurrentWidget != null)
        {
            CurrentWidget.RemoveFromViewport();
            CurrentWidget = null;
        }
        if (NewWidgetClass != null)
        {
            CurrentWidget = CreateWidget<UUserWidget>(GetWorld(), NewWidgetClass,"");
            if (CurrentWidget != null)
            {
                CurrentWidget.AddToViewport(0);
            }
        }
    }

}